vazhin/app.js

```diff
@@ -0,0 +1,283 @@
+            if (thisTaskDeadline <= today) {
+                deadline.classList.add("date-overdue");
+            }
+        }
+        checkDateOverdue();
+
+        trashBtn.addEventListener('click', function () {
+
+            arrOfTasks.splice(i, 1);
+            updateTasksInList();
+            if (todoList.childElementCount === 0) {
+                todoList.appendChild(noTasks);
+            }
+        })
+
+        let completedBtnClicked = false;
+        completedBtn.addEventListener('click', function () {
+
+            const motivatingWords = [
+                "you nailed that one 🙌",
+                "Keep going 😎",
+                "There you go!",
+                "Keep up the good work.",
+                "Keep it up.",
+                "Good job.",
+                "I’m so proud of you!",
+                "That was awesome!",
+                "Well done.",
+                "keep pushing!",
+                "You can do this!",
+                "That's how we do it 😎",
+                "Well done.",
+                "Never give up, just like that.",
+                "Do not stop 🙌",
+                "You're amazing!",
+                "You're the best",
+                "Well, that was impressive.",
+                "It's possible! don't forget that.",
+                "amazing job 💯"
+            ]
+            const motivatingWordsNode = document.createElement('p');
+            let randomIndex = Math.floor(Math.random() * motivatingWords.length);
+            motivatingWordsNode.innerText = `${motivatingWords[randomIndex]}`
+            motivatingWordsNode.style.textAlign = "center";
+            motivatingWordsNode.classList.add("motivating");
+
+            if (completedBtnClicked == false) {
+
+                task.innerHTML = `<strike>task: ${arrOfTasks[i].task}</strike>`;
+                priority.remove();
+                deadline.remove();
+                todoDiv.insertBefore(motivatingWordsNode, todoDiv.childNodes[1]);
+                completedBtn.innerHTML = '<i class="fas fa-check"></i>';
+                completedBtnClicked = true;
+                return;
+            }
+            else if (completedBtnClicked == true) {
+
+                task.innerHTML = `task: ${arrOfTasks[i].task}`;
+                completedBtn.innerHTML = '<i></i>';
+                document.querySelector('.motivating').remove();
+                todoDiv.insertBefore(priority, todoDiv.childNodes[1]);
+                todoDiv.insertBefore(deadline, todoDiv.childNodes[2]);
+                completedBtnClicked = false;
+                return;
+            }
+        })
+
+        editBtn.addEventListener("click", function () {
+            inEditMode = true;
+            indexOfInEditTask = i;
+
+            addBtn.innerHTML = "edit task"
+            taskText.value = arrOfTasks[i].task;
+            priorityInput.value = arrOfTasks[i].priority;
+            deadlineInput.value = arrOfTasks[i].deadline;
+        })
+    }
```

Comment:
While this for-loop is awesome, it got many, many things to do and while it is ok sometimes, some tweaks can be added to make the for-loop easier on CPU and better for readability.